matcher/cache/cache.go

```diff
@@ -364,18 +361,20 @@ func (c *cache) promote(now time.Time, elem *element) {
 	c.metrics.promotions.Inc(1)
 }
 
-func (c *cache) invalidateWithLock(nsHash xid.Hash, idHash xid.Hash, results results) results {
+func (c *cache) invalidateWithLock(namespace, id []byte, results results) results {
 	if c.invalidationMode == InvalidateAll {
 		c.toDelete = append(c.toDelete, results.elems)
 		c.notifyDeletion()
-		results.elems = make(elemMap)
-		c.namespaces[nsHash] = results
+		results.elems = newElemMap(elemMapOptions{})
+		c.namespaces.Set(namespace, results)
 	} else {
-		elem := results.elems[idHash]
-		delete(results.elems, idHash)
-		c.list.Lock()
-		c.list.Remove(elem)
-		c.list.Unlock()
+		elem, ok := results.elems.Get(id)
```

Comment:
Don't think you need to check `ok` here, `id` is guaranteed to be in the map when if `invalidateWithLock` is called and we get here.

Reply:
Sure thing.